controller/settingscontroller.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * ownCloud - Richdocuments App
+ *
+ * @author Blagovest Petrov
+ * @copyright 2015 Blagovest Petrov <blagovest@petrovs.info>
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later.
+ */
+
+namespace OCA\TwoFactor_Sms\Controller;
+
+use \OCP\AppFramework\Controller;
+use \OCP\IConfig;
+use \OCP\IRequest;
+use \OCP\IL10N;
+use \OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\TemplateResponse;
+
+//use OCA\Richdocuments\AppConfig;
+//use OCA\Richdocuments\Filter;
+
+class SettingsController extends Controller{
+
+	private $userId;
+	private $l10n;
+	private $appConfig;
+  private $appName = 'TwoFactor_Sms'
+
+	public function __construct($appName, IRequest $request, IL10N $l10n, AppConfig $appConfig, $userId){
```

Comment:
change ``AppConfig`` to ``IConfig`` and ``$useId`` to ``$UserId``, then dependency injection should be able to create the controller automatically.